Index: test-cases/dsm/testcases/ldt/DSM-2830/enable_ldt.py
===================================================================
--- test-cases/dsm/testcases/ldt/DSM-2830/enable_ldt.py	(revision 0)
+++ test-cases/dsm/testcases/ldt/DSM-2830/enable_ldt.py	(working copy)
@@ -0,0 +1,46 @@
+from selenium.common.exceptions import NoSuchElementException
+import unittest, time, sys
+from gui.constants import *
+from gui.modules import modules
+import lib
+from lib.workflowLib import WebDriver
+
+class Enable_LDT(unittest.TestCase):
+    def __init__(self, testname=None, values=None):
+        super(Enable_LDT, self).__init__(testname)
+        self.values = values
+
+    def setUp(self):
+        self.driver = WebDriver().Driver
+
+    def test_enable_ldt(self):
+        os.environ.update(self.values)
+        driver = self.driver
+        mod = modules.common_operations(driver)
+        mod.refresh_page()
+        try:
+            ldt = os.environ['LDT']
+            host_name = os.environ['AGENT_HOST']
+        except KeyError as e:
+            print "[ERROR] unable to find %s" % e
+            sys.exit(1)
+        try:
+            driver.find_element_by_link_text(hosts_link_text).click()
+            time.sleep(1)
+            driver.find_element_by_link_text(host_name).click()
+            time.sleep(1)
+            driver.find_element_by_xpath(d4_3_1_1_ldt_checkbox).click()
+            driver.find_element_by_xpath(d4_3_1_1_apply_bt).click()
+
+            if driver.find_element_by_xpath(d4_3_1_1_ldt_checkbox).is_selected():
+                print "[PASS] LDT is successfully Enable for the fs_agent %s." % host_name 
+            else:
+                print "[FAIL] failed to Enable LDT for the fs_agent %s." % host_name 
+
+        except NoSuchElementException as e:
+            print "[FAIL] Failed to enable LDT to host [%s]" % host_name
+            print "[ERROR] %s" % e
+            sys.exit(1)
+            
+    def tearDown(self):
+       self.driver.quit()
Index: test-cases/gui/constants.py
===================================================================
--- test-cases/gui/constants.py	(revision 67082)
+++ test-cases/gui/constants.py	(working copy)
@@ -1184,7 +1184,8 @@
 d4_1_ids_col = "table.dataview>tbody>tr>td:nth-child(7)>input"
 d4_1_key_col = "table.dataview>tbody>tr>td:nth-last-child(10)>input"
 d4_1_license_col = "table.dataview>tbody>tr>td:nth-last-child(5)>span"
-d4_1_push_col = "table.dataview>tbody>tr>td:nth-last-child(19)>span"
+#d4_1_push_col = "table.dataview>tbody>tr>td:nth-last-child(19)>span"
+d4_1_push_col = "//span[text()='Done']"
 #d4_1_docker_cb = "table.dataview>tbody>tr>td:nth-last-child(3)>input"
 d4_1_docker_cb = "//label[text()='Docker']/../..//input[@type='checkbox'][contains(@name, 'fragment1')]"
 d4_1_secure_start_cb = "//label[text()='Secure Start GuardPoint']/../..//input"
@@ -1270,6 +1271,7 @@
         /table/tbody/tr[5]/td[4]/span"
 d4_3_1_1_regen_chkbox = "//form[contains(@action,'cguctab')]/fieldset\
         /table/tbody/tr[7]/td[4]/span"
+d4_3_1_1_ldt_checkbox = "//label[text()='Live Data Transformation']/../../td[4]/span/input"
 
 #chkboxes of registration
 #d4_3_1_1_fs_reg = "//form[contains(@action,'cguctab')]/fieldset\
