Index: Application/DSM/DSMPages/BasePage.py
===================================================================
--- Application/DSM/DSMPages/BasePage.py	(revision 73467)
+++ Application/DSM/DSMPages/BasePage.py	(working copy)
@@ -56,8 +56,7 @@
         self.wait_for_menubar_to_load()
         self.EF.element_click(locator=self.locator_dict['dashboard'])
         self.EF.wait_for_element(
-            locator=
-            self.locator_dict['dashboard_vormetric_data_security_label'])
+            locator=self.locator_dict['dashboard_vormetric_data_security_label'])
 
     def go_to_domain_page(self):
         """
@@ -68,7 +67,8 @@
         self.EF.wait_for_element(locator=self.locator_dict['domain'])
         self.EF.hover_over_an_element(locator=self.locator_dict['domain'])
         self.EF.element_click(locator=self.locator_dict['manage_domains'])
-        self.EF.wait_for_element(locator=self.locator_dict['manage_domain_txt'])
+        self.EF.wait_for_element(
+            locator=self.locator_dict['manage_domain_txt'])
 
     def go_to_admin_page(self):
         """
@@ -77,7 +77,8 @@
         """
         self.wait_for_menubar_to_load()
         self.EF.element_click(locator=self.locator_dict['administrator'])
-        self.EF.wait_for_element(locator=self.locator_dict['administrator_title_txt'])
+        self.EF.wait_for_element(
+            locator=self.locator_dict['administrator_title_txt'])
 
     def go_to_high_availability_page(self):
         """
@@ -87,7 +88,8 @@
 
         self.wait_for_menubar_to_load()
         self.EF.element_click(locator=self.locator_dict['high_availability'])
-        return self.EF.wait_for_element(locator=self.locator_dict['high_availability_page_title'])
+        return self.EF.wait_for_element(
+            locator=self.locator_dict['high_availability_page_title'])
 
     def go_to_host_page(self):
         """
@@ -109,7 +111,8 @@
         self.wait_for_menubar_to_load()
         self.EF.hover_over_an_element(locator=self.locator_dict['hosts'])
         self.EF.element_click(locator=self.locator_dict['host_host_group'])
-        return self.EF.wait_for_element(locator=self.locator_dict['host_host_group_title'])
+        return self.EF.wait_for_element(
+            locator=self.locator_dict['host_host_group_title'])
 
     def go_to_registration_shared_secret_page(self):
         """
@@ -137,9 +140,11 @@
         self.EF.hover_over_an_element(
             locator=self.locator_dict['host_registration_shared_secret'])
         self.EF.element_click(
-            locator=self.locator_dict['host_registration_shared_secret_default_license_type'])
+            locator=self.locator_dict[
+                'host_registration_shared_secret_default_license_type'])
         return self.EF.wait_for_element(
-            self.locator_dict['host_registration_shared_secret_default_license_type_title'])
+            self.locator_dict[
+                'host_registration_shared_secret_default_license_type_title'])
 
     def go_to_host_password_challenge_response_page(self):
         """
@@ -165,7 +170,8 @@
         self.wait_for_menubar_to_load()
         self.EF.hover_over_an_element(locator=self.locator_dict['hosts'])
         self.EF.element_click(locator=self.locator_dict['host_qos_schedule'])
-        return self.EF.wait_for_element(locator=self.locator_dict['host_qos_schedules_title'])
+        return self.EF.wait_for_element(
+            locator=self.locator_dict['host_qos_schedules_title'])
 
     def go_to_keys_page(self):
         """
@@ -219,7 +225,8 @@
         self.wait_for_menubar_to_load()
         self.EF.hover_over_an_element(locator=self.locator_dict['keys'])
         self.EF.element_click(locator=self.locator_dict['keys_vault_keys'])
-        return self.EF.wait_for_element(locator=self.locator_dict['keys_vault_keys_title'])
+        return self.EF.wait_for_element(
+            locator=self.locator_dict['keys_vault_keys_title'])
 
     def go_to_keys_vault_keys_export_import_keys_page(self):
         """
@@ -247,7 +254,8 @@
         self.wait_for_menubar_to_load()
         self.EF.hover_over_an_element(locator=self.locator_dict['keys'])
         self.EF.element_click(locator=self.locator_dict['keys_keys_template'])
-        return self.EF.wait_for_element(locator=self.locator_dict['keys_key_templates_title'])
+        return self.EF.wait_for_element(
+            locator=self.locator_dict['keys_key_templates_title'])
 
     def go_to_keys_agent_objects_page(self):
         """
@@ -259,7 +267,8 @@
         self.wait_for_menubar_to_load()
         self.EF.hover_over_an_element(locator=self.locator_dict['keys'])
         self.EF.element_click(locator=self.locator_dict['keys_agent_objects'])
-        return self.EF.wait_for_element(locator=self.locator_dict['keys_agent_objects_title'])
+        return self.EF.wait_for_element(
+            locator=self.locator_dict['keys_agent_objects_title'])
 
     def go_to_keys_identities_page(self):
         """
@@ -271,7 +280,8 @@
         self.wait_for_menubar_to_load()
         self.EF.hover_over_an_element(locator=self.locator_dict['keys'])
         self.EF.element_click(locator=self.locator_dict['keys_identities'])
-        return self.EF.wait_for_element(locator=self.locator_dict['keys_agent_identities_title'])
+        return self.EF.wait_for_element(
+            locator=self.locator_dict['keys_agent_identities_title'])
 
     def go_to_certificates_page(self):
         """
@@ -378,7 +388,8 @@
         self.EF.hover_over_an_element(locator=self.locator_dict['policies'])
         self.EF.element_click(
             locator=self.locator_dict['policies_policy_wizard'])
-        return self.EF.wait_for_element(locator=self.locator_dict['policies_policy_wizard_title'])
+        return self.EF.wait_for_element(
+            locator=self.locator_dict['policies_policy_wizard_title'])
 
     def go_to_policies_export_import_policies_page(self):
         """
@@ -447,7 +458,8 @@
         self.EF.hover_over_an_element(locator=self.locator_dict['system'])
         self.EF.element_click(
             locator=self.locator_dict['system_log_preferences'])
-        return self.EF.wait_for_element(locator=self.locator_dict['system_log_preferences_title'])
+        return self.EF.wait_for_element(
+            locator=self.locator_dict['system_log_preferences_title'])
 
     def go_to_system_network_diagnostics_page(self):
         """
@@ -460,7 +472,8 @@
         self.EF.hover_over_an_element(locator=self.locator_dict['system'])
         self.EF.element_click(
             locator=self.locator_dict['system_network_diagnostics'])
-        return self.EF.wait_for_element(locator=self.locator_dict['system_network_diagonostics_title'])
+        return self.EF.wait_for_element(
+            locator=self.locator_dict['system_network_diagonostics_title'])
 
     def go_to_system_wrapper_keys_page(self):
         """
@@ -472,7 +485,8 @@
         self.wait_for_menubar_to_load()
         self.EF.hover_over_an_element(locator=self.locator_dict['system'])
         self.EF.element_click(locator=self.locator_dict['system_wrapper_keys'])
-        return self.EF.wait_for_element(locator=self.locator_dict['system_wrapper_keys_title'])
+        return self.EF.wait_for_element(
+            locator=self.locator_dict['system_wrapper_keys_title'])
 
     def go_to_system_backup_restore_page(self):
         """
@@ -485,7 +499,8 @@
         self.EF.hover_over_an_element(locator=self.locator_dict['system'])
         self.EF.element_click(
             locator=self.locator_dict['system_backup_restore'])
-        return self.EF.wait_for_element(locator=self.locator_dict['system_backup_restore_title'])
+        return self.EF.wait_for_element(
+            locator=self.locator_dict['system_backup_restore_title'])
 
     def go_to_system_backup_restore_auto_backup_page(self):
         """
@@ -514,7 +529,8 @@
         self.EF.hover_over_an_element(locator=self.locator_dict['system'])
         self.EF.element_click(
             locator=self.locator_dict['system_software_upgrade'])
-        return self.EF.wait_for_element(locator=self.locator_dict['system_upgrade_software_title'])
+        return self.EF.wait_for_element(
+            locator=self.locator_dict['system_upgrade_software_title'])
 
     def go_to_system_software_upgrade_patch_list_page(self):
         """
@@ -543,7 +559,8 @@
         self.EF.hover_over_an_element(locator=self.locator_dict['system'])
         self.EF.element_click(
             locator=self.locator_dict['system_intermediate_ca'])
-        return self.EF.wait_for_element(locator=self.locator_dict['system_intermediate_ca_title'])
+        return self.EF.wait_for_element(
+            locator=self.locator_dict['system_intermediate_ca_title'])
 
     def go_to_system_webserver_certi_page(self):
         """
@@ -570,7 +587,8 @@
         self.EF.hover_over_an_element(locator=self.locator_dict['system'])
         self.EF.element_click(
             locator=self.locator_dict['system_upload_rsa_configuration_file'])
-        return self.EF.wait_for_element(locator=self.locator_dict['system_upload_rsa_title'])
+        return self.EF.wait_for_element(
+            locator=self.locator_dict['system_upload_rsa_title'])
 
     def go_to_system_snmp_page(self):
         """
@@ -595,7 +613,8 @@
         self.EF.hover_over_an_element(locator=self.locator_dict['system'])
         self.EF.element_click(
             locator=self.locator_dict['system_email_notification'])
-        return self.EF.wait_for_element(locator=self.locator_dict['system_email_notification_title'])
+        return self.EF.wait_for_element(
+            locator=self.locator_dict['system_email_notification_title'])
 
     def go_to_system_ldap_page(self):
         """
@@ -648,12 +667,14 @@
 
         if validate_username:
             if domain == "":
-                self.EF.wait_for_element(locator=self.locator_dict['logged_in_label'])
+                self.EF.wait_for_element(
+                    locator=self.locator_dict['logged_in_label'])
             else:
                 self.EF.wait_for_element(
                     self.locator_dict['logged_in_domain_label'])
         else:
-            self.EF.wait_for_element(locator=self.locator_dict['logged_in_label'])
+            self.EF.wait_for_element(
+                locator=self.locator_dict['logged_in_label'])
 
     def logout(self, validate=False, print_exception=True):
         """
@@ -746,11 +767,15 @@
 
     def wait_for_menubar_to_load(self, wait_time=dsm_const.GLOBAL_WAIT_TIME):
         time.sleep(wait_time)
+        self.wait_for_wicket_div()
+
+    def wait_for_wicket_div(self, wait_time=dsm_const.GLOBAL_WAIT_TIME):
         wicket_div_flag = True
         while wicket_div_flag:
             try:
-                (element, by_type, locator_value) =  self.EF.wait_for_element(
+                (element, by_type, locator_value) = self.EF.wait_for_element(
                     locator=self.locator_dict['wicket_div'],
+                    timeout=wait_time,
                     print_exception=False)
             except Exception:
                 wicket_div_flag = False
Index: Application/DSM/DSMPages/DomainPage.py
===================================================================
--- Application/DSM/DSMPages/DomainPage.py	(revision 73467)
+++ Application/DSM/DSMPages/DomainPage.py	(working copy)
@@ -280,6 +280,8 @@
             self.search_domain(domain_name=domain_name,
                                print_info_flag=print_info_flag,
                                print_exception=print_exception)
+
+            # method implemented in base bage
             self.EF.element_click(
                 locator=self.locator_dict['manage_domain_domain_name_in_table'],
                 value_tuple=(domain_name),
@@ -901,6 +903,8 @@
             (element, by_locator_type, locator) = self.EF.wait_for_element(
                 locator=self.locator_dict['manage_domain_domain_name_in_table'],
                 value_tuple=(domain_name))
+
+            self.wait_for_wicket_div()
             if return_domain_element:
                 return element
             elif element:
Index: Tests/DSM/AdminPage/testlist
===================================================================
--- Tests/DSM/AdminPage/testlist	(revision 73467)
+++ Tests/DSM/AdminPage/testlist	(working copy)
@@ -1,2 +1,4 @@
 test_CreateDeleteAdmin.py
-test_EditAdminDetails.py
\ No newline at end of file
+test_EditAdminDetails.py
+test_AddRemoveAdminFromDomain.py
+test_EnableDisableAdmin.py
\ No newline at end of file
Index: Tests/DSM/DomainPage/test_CreateDeleteDomainByGui.py
===================================================================
--- Tests/DSM/DomainPage/test_CreateDeleteDomainByGui.py	(revision 73467)
+++ Tests/DSM/DomainPage/test_CreateDeleteDomainByGui.py	(working copy)
@@ -62,12 +62,12 @@
 
         """ Step 2. Add domain. """
         self.domain_page_object.create_domain(
-            domain_dict=domain_dict, by_method=pom_const.TEST_BY_METHOD)
+            domain_dict=domain_dict)
 
         """ Step 3. Delete domain. """
 
         self.domain_page_object.delete_domain(
-            domain_name=domain_dict['name'], by_method=pom_const.TEST_BY_METHOD,
+            domain_name=domain_dict['name'],
             print_exception=False)
         RB.print_info(
             text="test_CreateDeleteDomainByGui passed successfully..")
